### ISSUE TITLE
Preserve async result provenance across state-machine fields

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#5025]
Parent tracker: #4611
Discovered during: #5021 / #4790
Prerequisite for complete lowering-independent JSON-wire facts in: #4572

## Owner

`ILInspector.Analysis`, specifically result-sink value provenance across authenticated compiler async state machines.

## Problem

Equivalent C# can currently issue different `MethodResultSink` evidence across compiler async and .NET 11 runtime async when a serialized result is stored before a suspension and returned afterward:

```csharp
[JSExport]
public static async Task<string> SerializeBeforeAwait(string name)
{
    string payload = JsonSerializer.Serialize(
        new WidgetDto(name, 0, [], null),
        FixtureJsonContext.Default.WidgetDto);
    await Task.Yield();
    return payload;
}
```

With compiler async, the authenticated `MoveNext` completion sink reads the hoisted payload from a state-machine field. Analysis reports the sink as incomplete with no direct-call source. With runtime async, the exported method's physical `ret` retains the serializer call as a complete source. `ILInspector.JsExportSurface` therefore leaves compiler `ReturnWireType` unset but resolves the runtime form.

This is missing Analysis evidence, not a reason for JsExportSurface or a TypeScript consumer to recognize lowering-specific IL shapes.

## Direction

- Extend Analysis-owned result provenance so an authenticated compiler-state-machine completion can retain a value stored before suspension and loaded from the corresponding state-machine field afterward.
- Preserve exact field, physical body, kickoff source, and lowering identities.
- Do not add serializer, JSExport, display-name, or generated-field-name heuristics to Analysis.
- Keep ambiguous stores, foreign fields, unresolved paths, and malformed state-machine ownership incomplete.

## Acceptance

- Paired compiled runtime-async-off/on fixtures use the same genuinely awaited method that serializes before suspension and returns the stored payload afterward.
- The compiler `MoveNext` sink and runtime exported-method sink both expose equivalent complete serializer provenance through an Analysis-owned typed contract.
- Close negatives reject an unrelated field store, multiple reaching field stores, a store after a different suspension path, and a lookalike or unauthenticated state-machine field.
- `ILInspector.JsExportSurface` consumes the owner-issued fact without a lowering-specific field heuristic and issues structurally equal `JsExportFunction` wire facts.

## Reproduction

Reproduced on #5021 exact head `eb3071adc846abf6463686657f5299980e34ad3b` with .NET SDK `11.0.100-preview.7.26366.102` in Release:

- compiler async: `SingleArgumentCall` in `MoveNext`, `StateMachine`, incomplete, zero source calls, `ReturnWireType = null`;
- runtime async: `MethodReturn` in the exported method, `Runtime`, complete, serializer source retained, `ReturnWireType = WidgetDto`.
